### PR TITLE
Update api version for internal apis to v1alpha1

### PR DESCRIPTION
**Pull request by VaishnaviHire**
## Description

Should  be merged after #1389 

Review [commit](https://github.com/opendatahub-io/opendatahub-operator/commit/3af04a240890159f36d9307a87658c39214f0408)

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -103,15 +103,16 @@ metadata:
...
     olm.skipRange: '>=1.0.0 <2.19.0'
     operators.operatorframework.io/builder: operator-sdk-v1.31.0
     operators.operatorframework.io/internal-objects: '["featuretrackers.features.opendatahub.io",
       "codeflares.components.opendatahub.io", "dashboards.components.opendatahub.io",
       "datasciencepipelines.components.opendatahub.io", "kserves.components.opendatahub.io",
       "kueues.components.opendatahub.io", "modelmeshservings.components.opendatahub.io",
       "modelregistries.components.opendatahub.io", "rays.components.opendatahub.io",
-      "trainingoperators.components.opendatahub.io", "trustyais.components.opendatahub.io",  "workbenches.components.opendatahub.io"]'
+      "trainingoperators.components.opendatahub.io", "trustyais.components.opendatahub.io",  "workbenches.components.opendatahub.io",
+      "monitorings.services.opendatahub.io" ]'
```
Comment (ykaliuta) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:115]: is it part of #1389?

File: controllers/components/codeflare/codeflare.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	componentsv1alpha1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1alpha1"
```
Comment (ykaliuta) [inline controllers/components/codeflare/codeflare.go:10]: @lburgazzoli I'm wondering while there is no dependency of 2 versions simultaneously in the code, why to put the version in the namespace? It causes such huge renaming like this for example.

Comment (lburgazzoli) [reply controllers/components/codeflare/codeflare.go:10]: agree

FWIW, maybe we can use something like `componentsApi` to point to the storage version of the API is there is a need to have an alias. So we would have a stable naming for the actual API and aliases for alternatives if needed.

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (lburgazzoli) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:115]: I think so, the descriptions mentions: _Should be merged after https://github.com/opendatahub-io/opendatahub-operator/pull/1389_

**Comment by lburgazzoli** [comment]
@VaishnaviHire wonder if we also want to move the group to `operator.opendatahub.io` or `platform.opendatahub.io`

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (ykaliuta) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:115]: > I think so, the descriptions mentions: _Should be merged after #1389_

it says, but it is a part of the patch 3af04a240890 ("Update api version for internal apis to v1alpha1")

Comment (zdtsw) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:115]: we should aim to get the monitoring PR in first,  this one included the other, will need a big rebase eventually.

Comment (ykaliuta) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:115]: IIUC it's supposed to be one patch on top on the #1389 So, not such a big rebase

File: controllers/components/codeflare/codeflare.go (hunk 2 above)
Comment (VaishnaviHire) [reply controllers/components/codeflare/codeflare.go:10]: I am okay with updating it to `componentsApi`. I agree keeping names version agnostic is helpful.

**Comment by VaishnaviHire** [comment]
> @VaishnaviHire wonder if we also want to move the group to `operator.opendatahub.io` or `platform.opendatahub.io` to avoid conflicts

So this should just include annotations and PartOfLabel?

**Comment by lburgazzoli** [comment]
> > @VaishnaviHire wonder if we also want to move the group to `operator.opendatahub.io` or `platform.opendatahub.io` to avoid conflicts
> 
> So this should just include annotations and PartOfLabel?

Not only, also the actual group of the component api https://github.com/opendatahub-io/opendatahub-operator/blob/feature-operator-refactor/apis/components/v1/groupversion_info.go#L29

**Comment by VaishnaviHire** [comment]
Closing this PR and opening another one. Its not reflecting the latest commits

**Pull request closed without merging**